src/marionette.rs

```diff
@@ -380,18 +380,35 @@ fn unzip_buffer(buf: &[u8], dest_dir: &Path) -> WebDriverResult<()> {
                 "Processing zip file failed")
         }));
         let unzip_path = {
-            let rel_path = Path::new(file.name());
-            let unzip_path = dest_dir.join(rel_path);
-            if let Some(dir) = unzip_path.parent() {
-                if !dir.exists() {
-                    try!(fs::create_dir_all(dir));
+            let name = file.name();
+            let is_dir = name.ends_with("/");
```

Comment:
Well the directory doesn't actually exist yet, so `Path.is_dir()` will always be `false`.